src/libs/ReportActionsUtils.ts

```diff
@@ -214,15 +214,53 @@ function getSortedReportActions(reportActions: ReportAction[] | null, shouldSort
         return (first.reportActionID < second.reportActionID ? -1 : 1) * invertedMultiplier;
     });
 
-    // If shouldMarkTheFirstItemAsNewest is true, label the first reportAction as isNewestReportAction
-    if (shouldMarkTheFirstItemAsNewest && sortedActions?.length > 0) {
-        sortedActions[0] = {
-            ...sortedActions[0],
-            isNewestReportAction: true,
-        };
+    return sortedActions;
+}
+
+/**
+ * Returns the range of report actions from the given array which include current id
+ * the range is consistent
+ *
+ * param {ReportAction[]} array
+ * param {String} id
+ * returns {ReportAction}
+ */
+function getRangeFromArrayByID(array: ReportAction[], id?: string): ReportAction[] {
+    let index;
+
+    if (id) {
+        index = array.findIndex((obj) => obj.reportActionID === id);
+    } else {
+        index = array.findIndex((obj) => obj.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
```

Comment:
This piece of functionality to default for looking for the first non-pending action is kind of hiding here. Not sure if it should live here or not 🤔